roles/federation-demo/tasks/deploy.yml

```
@@ -0,0 +1,83 @@
+---
+  - name: Check if project exists
+    stat:
+      path: "{{ federation_git.clone_dest }}"
+    register: project_stat
+
+  - name: Clean clone directory
+    file:
+      state: absent
+      path: "{{ federation_git.clone_dest }}"
+    when: project_stat.stat.exists
+
+  - name: Clone the repository
+    git:
+      repo: "{{ federation_git.git_repo }}"
+      version: "{{federation_git.git_branch}}"
+      dest: "{{federation_git.clone_dest}}"
+      key_file: "{{ federation_git.key_path }}"
+
+
+  - name: Install node modules for react
+    npm:
+      path: "{{federation_react_path}}"
+
+  - name: Install node modules for express
+    npm:
+      path: "{{federation_express_path}}"
+
+  - name: Install node modules for ams-agent
+    npm:
+      path: "{{federation_ams_agent_path}}"
+
+  - name: React config
+    template:
+      src: react-config.j2
+      dest: "{{federation_react_path}}/src/config.json"
```

```suggestion
      dest: "{{federation_react_path}}/src/config.json"
      backup: yes
```

I suggest you use the `backup` parameter on most of the time ( template/file/copy/lineinfile )
https://docs.ansible.com/ansible/2.9/modules/template_module.html#parameter-backup